optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/employee/EmployeeListPanel.java

```diff
@@ -37,19 +31,24 @@
 import org.jboss.errai.ui.shared.api.annotations.EventHandler;
 import org.jboss.errai.ui.shared.api.annotations.Templated;
 import org.optaplanner.openshift.employeerostering.gwtui.client.common.FailureShownRestCallback;
-import org.optaplanner.openshift.employeerostering.gwtui.client.popups.ErrorPopup;
+import org.optaplanner.openshift.employeerostering.gwtui.client.common.ValidationListener;
+import org.optaplanner.openshift.employeerostering.gwtui.client.common.Validator;
 import org.optaplanner.openshift.employeerostering.gwtui.client.resources.css.CssResources;
+import org.optaplanner.openshift.employeerostering.shared.employee.Employee;
+import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeRestServiceBuilder;
 import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeSkillProficiency;
 import org.optaplanner.openshift.employeerostering.shared.skill.Skill;
 import org.optaplanner.openshift.employeerostering.shared.skill.SkillRestServiceBuilder;
-import org.optaplanner.openshift.employeerostering.shared.employee.Employee;
-import org.optaplanner.openshift.employeerostering.shared.employee.EmployeeRestServiceBuilder;
 import org.optaplanner.openshift.employeerostering.shared.tenant.Tenant;
 
-import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.*;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.General_actions;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.General_delete;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.General_edit;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.General_name;
+import static org.optaplanner.openshift.employeerostering.gwtui.client.resources.i18n.OptaShiftUIConstants.General_skills;
 
 @Templated
-public class EmployeeListPanel implements IsElement {
+public class EmployeeListPanel implements IsElement, ValidationListener {
```

Comment:
Okay, now that I see the usage, I think you are not fully taking advantage of the tools available for validation in GWT and Errai.

First, GWT comes with (partially) declarative [validation](http://www.gwtproject.org/doc/latest/DevGuideValidation.html). Errai adds a bit of [extra support](http://docs.jboss.org/errai/latest/errai/reference/html_single/#sid-51282340_DataBinding-Beanvalidation) on top, such as allowing you to inject a GWT validator.

This feature pairs well with [Errai Data-Binding](http://docs.jboss.org/errai/latest/errai/reference/html_single/#sid-51282340). Since you are already using `@Templated` beans, it's easy to use Errai data-binding declaratively.

Here is how you could add validation to this bean with Errai/GWT validation:

1. Define a model for your view (EmployeeList).
2. Add constraints on your model (in some advanced cases, you might need to create your own constraints, but the basics should be covered).
3. Use data-binding to bind model fields to respective UI fields.
4. Add a property-change handler (there is an annotation for declaring a method), and call the GWT Validator from here.

Errai data-binding is very easy to use for simple models, but does have some thorns when working with lists/maps. But IMO it's better for you to use more of the tech in our platform and possibly contribute feedback/fixes when you encounter problems than to re-implement something like validation unless there is a *very* compelling reason.